src/heartbeat.py

```diff
@@ -0,0 +1,52 @@
+import aiohttp
+import threading
+import time
+import asyncio
+
+
+class Heartbeat:
+    def __init__(self, base_url, interval_ms):
+        self.base_url = base_url
+        self.running = True
+        self.thread = None
+        self.interval_ms = interval_ms
+
+    async def start(self, task_id):
+        try:
+            heartbeat_url = f'{self.base_url}/{task_id}'
+            self.thread = threading.Thread(target=self.thread_callback, args=[heartbeat_url])
+            self.thread.start()
+        except Exception as e:
+            print(f'Error occurred: {e}.')  # TODO: replace by mc-logger
+            raise e
+
+    def stop(self):
+        try:
+            if self.running is True:
+                self.running = False
+                print('stopping thread')  # TODO: replace by mc-logger
```

Comment:
better log on what thread is being closed (Stopping heartbeat)

Reply:
fixed in 
4c17adc